examples/remote/binary_dependencies/cargo/crates.bzl

```diff
@@ -9,6 +9,144 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")  # bui
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")  # buildifier: disable=load
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")  # buildifier: disable=load
 
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of normal dependencies for the Rust targets of that package.
+_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+        "ferris-says": "@remote_binary_dependencies__ferris_says__0_2_0//:ferris_says",
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of proc_macro dependencies for the Rust targets of that package.
+_PROC_MACRO_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of normal dev dependencies for the Rust targets of that package.
+_DEV_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of proc_macro dev dependencies for the Rust targets of that package.
+_DEV_PROC_MACRO_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+def crates(deps):
+    """EXPERIMENTAL -- MAY CHANGE AT ANY TIME: Finds the fully qualified label of the requested crates for the package where this macro is called.
+
+    WARNING: This macro is part of an expeirmental API and is subject to change.
+
+    Args:
+        deps (list or str): Either a list of dependencies or a string of one which will
+            be converted into a list.
+    Returns:
+        list: A list of labels to cargo-raze generated targets (str)
+    """
+
+    # Join both sets of dependencies
+    dependencies = dict()
+    for dep_map in [_DEPENDENCIES, _PROC_MACRO_DEPENDENCIES, _DEV_DEPENDENCIES, _DEV_PROC_MACRO_DEPENDENCIES]:
+        for package_name in _DEPENDENCIES:
+            if package_name in dependencies:
+                dependencies[package_name].extend(dep_map[package_name])
+            else:
+                dependencies[package_name].update(dep_map[package_name])
+
+    if not deps:
+        fail("An invalid argument has been provided. Please pass a crate name or a list of crate names")
```

Comment:
Does this need to be an error? I'd expect empty list to return empty list.

Reply:
What's the use case of passing an empty list? I think that's a better assumption but I was trying to be more defensive here and not allow unknown behavior before users decide they needed for less than ideal reasons.

Reply:
If you can operate with the input (even if the return value is not useful) it's best to just deal with it, rather than break.